### ISSUE TITLE
`Process.Modules` is unable to list 32-bit modules of a target process on Windows

### ISSUE THREAD

**Issue by just-ero** [dotnet/runtime#70370]
### Description

`Process.Modules` is unable to list 32-bit modules of a 32-bit target process when the calling process is 64-bit.

### Reproduction Steps

A running 32-bit process is required. Using the `Process` class, find said process and enumerate its modules using the `Process.Modules` property.

### Expected behavior

All modules, 64-bit as well as 32-bit ones, should be listed by `Process.Modules`.

### Actual behavior

The listed modules will consist of only the main module and any 64-bit modules that are loaded by the process. No 32-bit modules will be listed.

### Regression?

To my knowledge, this has never worked. The lowest version I've had to deal with this on was .NET Framework 4.6.1.

### Known Workarounds

Usage of [`EnumProcessModulesEx`](https://docs.microsoft.com/en-us/windows/win32/api/psapi/nf-psapi-enumprocessmodulesex) instead of the current EnumProcessModules ensures the inclusion of all modules of any process (when the calling process is 64-bit).

Additionally, [`Module32FirstW`](https://docs.microsoft.com/en-us/windows/win32/api/tlhelp32/nf-tlhelp32-module32firstw) and [`Module32NextW`](https://docs.microsoft.com/en-us/windows/win32/api/tlhelp32/nf-tlhelp32-module32nextw) can be used to achieve the same thing.

### Configuration

Last tested on .NET 6, x64, Windows 11 Pro.  
The issue may not apply to 32-bit calling processes, which can only list modules of other 32-bit target processes anyway.

### Other information

[`NtProcessManager.GetModules()`](https://github.com/dotnet/runtime/blob/eef6c7f6cb7f73163313d349fbb796f37b4c3996/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs#L67) uses [`Interop.Kernel32.EnumProcessModules()`](https://github.com/dotnet/runtime/blob/eef6c7f6cb7f73163313d349fbb796f37b4c3996/src/libraries/Common/src/Interop/Windows/Kernel32/Interop.EnumProcessModules.cs#L14):

https://github.com/dotnet/runtime/blob/eef6c7f6cb7f73163313d349fbb796f37b4c3996/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs#L81

This is the reason only the main module and modules of the same architecture as the calling process can be listed.

**Comment by adamsitnik** [comment]
Hi @just-ero ! Is there any chance you would like to work on a fix?

**Comment by just-ero** [comment]
Hi! I wouldn't know the first thing about how I would do that!

**Comment by adamsitnik** [comment]
> I wouldn't know the first thing about how I would do that!

It's "simple" when you know how:

1. You fork the repo by clicking the fork button in right top corner
2. You clone your own fork and change the current directory to the folder where it got downloaded

```cmd
git clone https://github.com/just-ero/runtime.git
cd runtime
```

3. You create a new branch

```cmd
git checkout -b issue70370
```

4. You install the build requirements listed here: https://github.com/dotnet/runtime/blob/main/docs/workflow/README.md#build-requirements
5. You build the repo by running the following command (full process is described [here](https://github.com/dotnet/runtime/blob/main/docs/workflow/building/libraries/README.md)):

```cmd
build.cmd -subset clr+libs+libs.tests -rc Release`
```

6. You apply the fixes (changing the sys-call as you have mentioned)
7. You build the library by running

```cmd
dotnet.cmd build .\src\libraries\System.Diagnostics.Process\System.Diagnostics.Process.sln
```

8 You run the tests:

```cmd
dotnet.cmd build /t:Test .\src\libraries\System.Diagnostics.Process\tests\System.Diagnostics.Process.Tests.csproj /p:Configuration=Debug
```

Once all tests are passing you push the changes, got to https://github.com/dotnet/runtime and click on the notification that detected new changes and asks you for creating a new PR.

**Comment by adamsitnik** [comment]
Fixed by #71218, again thanks a lot @just-ero !
